src/Bonsai.ML.Torch/Buffer.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Reactive.Linq;
+using TorchSharp;
+
+namespace Bonsai.ML.Torch;
+
+/// <summary>
+/// This operator collects incoming tensors into a buffer and concatenates them along the first dimension.
+/// </summary>
+/// <remarks>
+/// The operator maintains an internal buffer that accumulates incoming tensors until it reaches the specified count.
+/// When the buffer reaches the specified count, it is emitted as a single tensor. After emitting the buffer, the operator skips a specified number of incoming tensors before starting to fill the buffer again.
+/// </remarks>
+[Combinator]
+[Description("Buffers the incoming tensors and concatenates them into a single tensor along the first dimension.")]
+[WorkflowElementCategory(ElementCategory.Combinator)]
+public class Buffer
+{
+    private int _count = 1;
+    /// <summary>
+    /// Gets or sets the number of tensors to accumulate in the buffer before emitting.
+    /// </summary>
+    [Description("The number of tensors to accumulate in the buffer before emitting.")]
+    public int Count
+    {
+        get => _count;
+        set => _count = value <= 0
+            ? throw new ArgumentOutOfRangeException("Count must be greater than zero.")
+            : value;
+    }
+
+    private int _skip = 1;
+    /// <summary>
+    /// Gets or sets the number of tensors to skip after emitting the buffer.
+    /// </summary>
+    [Description("The number of tensors to skip after emitting the buffer.")]
+    public int Skip
+    {
+        get => _skip;
+        set => _skip = value < 0
+            ? throw new ArgumentOutOfRangeException("Skip must be non-negative.")
+            : value;
+    }
+
+    private torch.Tensor _buffer = null;
+    private int _current = 0;
+    private torch.Tensor _idxSrc = null;
+    private torch.Tensor _idxDst = null;
```

Comment:
We can come back to this later, but I'm finding it makes it harder for me to read a class when private variables are scattered throughout the class code.

Usually layout is grouped by member type, e.g. fields at the top, followed by properties, methods, etc. A few references:
  - [C# at Google Style Guide](https://google.github.io/styleguide/csharp-style.html#organization)
  - [Style Cop Analyzer](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1201.md)

The ordering of the groups is not really standard, and we haven't yet published an official style guide for Bonsai, but I think it does help to at least have them grouped.